src/main/java/com/amazonaws/encryptionsdk/kms/AwsKmsCmkId.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except
+ * in compliance with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.amazonaws.encryptionsdk.kms;
+
+import com.amazonaws.arn.Arn;
+import com.amazonaws.encryptionsdk.exception.MalformedArnException;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+
+import java.util.Objects;
+
+/**
+ * A representation of an AWS KMS Customer Master Key Identifier, which may be one either a
+ * key ID, key Amazon Resource Name (ARN), alias name, or alias ARN.
+ */
+public class AwsKmsCmkId {
```

Comment:
I like having this explicit type. Curious why we don't care about the "alias/" prefix any more though.

Reply:
Previously we cared about the alias/ prefix because if it wasn't an alias we would assume it was an ARN. We then decide to continue supporting "raw" key IDs, which could basically be anything, so that assumption no longer holds. So now I just check if it looks like an arn and thus will validate it, otherwise anything is allowed.